### ISSUE TITLE
Document testing options in more detail in the v16 migration guide

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#7399]
### Describe the documentation issue

I was following https://stylelint.io/migration-guide/to-16#plugins and I had some issues with getting the tests working.

The first suggested migration path doesn't actually work when following the guide step by step :

```css
 {
   "scripts": {
-    "test": "jest",
+    "test": "cross-env NODE_OPTIONS=\"--experimental-vm-modules --no-warnings\" jest --runInBand",
   }
 }
```

I think it could work but I don't have `cross-env` installed so that gave an error.
I didn't want to install `cross-env` because the README says that it is in maintenance mode and I didn't want to investigate further as that felt like a sidetrack.

This also made me look closer at the actual test command and I didn't really like depending on experimental flags.

----------

The second path suggests using `jest-light-runner` if a specific error is encountered

> If you get a segmentation fault while running the preset on Node.js 18, you can use [jest-light-runner](https://www.npmjs.com/package/jest-light-runner):

But that wasn't my error, so I ignored this.

----------

The third option is actual gold :

> Alternatively, you can try our new [stylelint-test-rule-node](https://www.npmjs.com/package/stylelint-test-rule-node) package that uses the test runner and assertions built into Node.js:

Adding this and dropping `jest` removed almost 300 dev dependencies.
It also just worked.

### What solution would you like to see?

Maybe a paragraph needs to be added about `cross-env`?
So that users know that it needs to be added?

------------------

Maybe `stylelint-test-rule-node` can be suggested as the first option to try?

To me this feels like the better path forward and if the first options hadn't failed I wouldn't have tried it.

**Comment by romainmenke** [comment]
This was the only point at which I felt a little bit stuck during the migration process.
Everything else just worked.

I followed the steps, got stuck a bit with the `test` command but I found an alternative path in the migration guides.

Everything just worked and a bunch of things became redundant so I got to delete a lot of stuff 🎉 

https://github.com/csstools/stylelint-media-use-custom-media/commit/a211b2ce662273bc52fe5876cb265bbd87d98318#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Comment by ybiquitous** [comment]
@romainmenke Thanks for the helpful feedback. Happy to hear your plugin now supports Stylelint v16 without big problems!

About Jest and `node:test`, there are still some troubles:

- Plugin authors need to rewrite their test code from Jest to `node:test` (especially, `expect()` -> `assert()`).
- `node:test` has been stable with Node.js 20. It was still experimental with Node.js 18.
- `node:test` still has an experimental [coverage support](https://nodejs.org/api/test.html#collecting-code-coverage).
- `stylelint-test-rule-node` may be still immature.
- `jest-light-runner` also has a limited [coverage support](https://github.com/nicolo-ribaudo/jest-light-runner#partially-supported-features).

Thus, the current migration guide about testing needs to provide several migration paths, and so the guide may confuse plugin creators.

But I agree with improving the guide. What do you think? @jeddy3

**Comment by jeddy3** [comment]
@romainmenke Thank you for sharing. 

Let's expand the testing part of the migration guide to better surface the nuances of each option so that users can choose which is best for them.

I can pick this up towards the end of the week.
